core/vm/contracts.go

```diff
@@ -121,6 +125,19 @@ var PrecompiledContractsBLS = map[common.Address]PrecompiledContract{
 	common.BytesToAddress([]byte{18}): &bls12381MapG2{},
 }
 
+var PrecompiledContracts4788 = map[common.Address]PrecompiledContract{
+	common.BytesToAddress([]byte{1}): &ecrecover{},
+	common.BytesToAddress([]byte{2}): &sha256hash{},
+	common.BytesToAddress([]byte{3}): &ripemd160hash{},
+	common.BytesToAddress([]byte{4}): &dataCopy{},
+	common.BytesToAddress([]byte{5}): &bigModExp{eip2565: true},
+	common.BytesToAddress([]byte{6}): &bn256AddIstanbul{},
+	common.BytesToAddress([]byte{7}): &bn256ScalarMulIstanbul{},
+	common.BytesToAddress([]byte{8}): &bn256PairingIstanbul{},
+	common.BytesToAddress([]byte{9}): &blake2F{},
+	params.BeaconRootsStorageAddress: &beaconRoot{},
```

Comment:
Given that all other contracts are hard coded with value addresses, I'd rather stick to that vs adding a param here

Reply:
This one is slightly different though as it needs to be accessed in both `consensus/misc/eip4788.go` and in `core/vm/contracts.go`. It's weird it is the only non-hardcoded one here though, but is it better to hardcode in both places?

Reply:
It should be consistent. If there's a need to move the beaconRoot into a param, then move all of them imo.